packages/riverpod/lib/src/framework/base_provider.dart

```diff
@@ -315,6 +316,11 @@ abstract class ProviderReference {
   /// care about this.
   bool get mounted;
 
+  /// Will return the currently exposed value of the provider.
+  ///
+  /// This is useful when dealing with asynchronous operations.
+  Listened get currentValue;
```

Comment:
I'd name it "state", to match with StateNotifier and a potential `setState`

I'd also document that it cannot be used within "build"